src/Adapter/Order/AbstractOrderHandler.php

```diff
@@ -44,7 +46,18 @@ abstract class AbstractOrderHandler
      */
     protected function getOrderObject(OrderId $orderId)
     {
-        $order = new Order($orderId->getValue());
+        try {
+            $order = new Order($orderId->getValue());
+        } catch (PrestaShopException $e) {
+            throw new OrderException(
```

Comment:
Should be OrderNotFoundException? :thinking:

Reply:
is this try/catch even necessary, given there's an OrderNotFoundException thrown at line 62 if we don't have the right Order object ?

Reply:
Ideally, we want to surround all Prestashop exceptions and replace by new core exceptions. Theoretically it will never happen i guess.